frontend/routes/signUp/signUp.component.tsx

```diff
@@ -0,0 +1,320 @@
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from 'react';
+import * as Yup from 'yup';
+import { Formik, Field, Form } from 'formik';
+
+import { getPasswordStrength, getPasswordStrengthMessage, schema } from '../../services/validation';
+import { clientConfigService } from './../../services/clientConfig';
+
+import { Panel } from '../components/panel/panel.component';
+import { Logo } from '../components/logo/logo.component';
+import { Footer } from './components/footer';
+import { SubmitButton } from '../components/submitButton/submitButton.component';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import { ReCaptcha } from './components/reCaptcha/reCaptcha.component';
+
+import {
+	Container,
+	Headline,
+	StyledGrid,
+	StyledFormControl,
+	ButtonContainer
+} from './signUp.styles';
+import { FieldsRow, StyledTextField } from '../profile/profile.styles';
+
+const RegistrationInitialValues = {
+	captcha: '',
+	company: '',
+	countryCode: '',
+	email: '',
+	emailConfirm: '',
+	username: '',
+	firstName: '',
+	lastName: '',
+	password: '',
+	passwordConfirm: '',
+	termsAgreed: false,
+	mailListAgreed: false
+};
+
+const RegistrationSchema = Yup.object().shape({
+	username: schema.required,
+	password: schema.password
+		.min(0)
+		.strength(1, 'This password is weak'),
+	passwordConfirm: schema.password
+		.min(0)
+		.equalTo(
+			Yup.ref('password'),
+			'Password confirmation must match password'
+		),
+	firstName: schema.required,
+	lastName: schema.required,
+	email: schema.email,
+	emailConfirm: schema.email
+		.equalTo(
+			Yup.ref('email'),
+			'Email confirmation must match email'
+		),
+	countryCode: schema.required,
+	captcha: clientConfigService.captcha_client_key ? schema.required : Yup.string(),
+	termsAgreed: Yup.boolean().oneOf([true])
+});
+
+const DEFAULT_INPUT_PROPS = {
+	margin: 'normal',
+	required: true
+};
+
+interface IProps {
+	history: any;
+	location: any;
+	headlineText?: string;
+	onRegister: (username, data) => void;
+	isPending: boolean;
+}
+
+interface IState {
+	passwordStrengthMessage: string;
+}
+
+export class SignUp extends React.PureComponent<IProps, IState> {
+	public state = {
+		passwordStrengthMessage: ''
+	};
+
+	public reCaptchaWrapperRef = React.createRef<any>();
+
+	public handleSubmit = (values, form) => {
+		const data = {
```

Comment:
Could this be changed to:
```suggestion
		const data = {...values};
```

Or there are more data that's being passed that you dont want?

Reply:
Good point! Exactly, I used lodash omit function for this one.